### PR TITLE
PVCORE-4213 - Initial tracing commit

### PR DESCRIPTION
**Pull request by mike-kingsbury**
Initial Tracing support in Konflux.  Supports Tekton pipeline tracing forwarded to a local OpenTelemetry collector.  Future work includes a local instance of Jaeger for visualization and forwarding to SaaS provider(s).

### CODE REVIEW

File: components/tracing/OWNERS
Comment (amisstea) [inline components/tracing/OWNERS:1]: There's nothing in this component directory. Do we need this?

File: argo-cd-apps/overlays/konflux-public-production/kustomization.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ namespace: konflux-public-production
 
 patchesStrategicMerge:
   - delete-applications.yaml
+  - delete-dev-only-applications.yaml
```
Comment (amisstea) [inline argo-cd-apps/overlays/konflux-public-production/kustomization.yaml:18]: Why not reuse `delete-applications.yaml` for the respective overlays and add the AppSet deletion patch in there? AFAIK it'll have the same effect.

File: argo-cd-apps/base/all-clusters/infra-deployments/enable-tekton-tracing/enable-tekton-tracing.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: enalbe-tekton-tracing
```
Comment (amisstea) [inline argo-cd-apps/base/all-clusters/infra-deployments/enable-tekton-tracing/enable-tekton-tracing.yaml:4]:
```suggestion
  name: enable-tekton-tracing
```

Comment (mike-kingsbury) [reply argo-cd-apps/base/all-clusters/infra-deployments/enable-tekton-tracing/enable-tekton-tracing.yaml]: Fixed

File: argo-cd-apps/overlays/konflux-public-production/kustomization.yaml (hunk 1 above)
Comment (mike-kingsbury) [reply argo-cd-apps/overlays/konflux-public-production/kustomization.yaml]: I was honestly on the fence about it....  I liked the clarity of the filename indicating what the effect was....  Updated to reuse deleted-applications.

**Comment by mftb** [comment]
> @mike-kingsbury: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> ci/prow/appstudio-hac-e2e-tests	[a0ecdf5](https://github.com/redhat-appstudio/infra-deployments/commit/a0ecdf531abf8d0226b6c6661e5a3ae1f7f5f85c)	[link](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/redhat-appstudio_infra-deployments/3740/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-hac-e2e-tests/1795546422901215232)	false	`/test appstudio-hac-e2e-tests`
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=redhat-appstudio&repo=infra-deployments&pr=3740). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:mike-kingsbury).

@amisstea  Should we retrigger those?

**Comment by amisstea** [comment]
> > @mike-kingsbury: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> > Test name	Commit	Details	Required	Rerun command
> > ci/prow/appstudio-hac-e2e-tests	[a0ecdf5](https://github.com/redhat-appstudio/infra-deployments/commit/a0ecdf531abf8d0226b6c6661e5a3ae1f7f5f85c)	[link](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/redhat-appstudio_infra-deployments/3740/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-hac-e2e-tests/1795546422901215232)	false	`/test appstudio-hac-e2e-tests`
> > [Full PR test history](https://prow.ci.openshift.org/pr-history?org=redhat-appstudio&repo=infra-deployments&pr=3740). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:mike-kingsbury).
> 
> @amisstea Should we retrigger those?

It's an optional test and the failures look unrelated so try a few times but otherwise don't worry about them.

File: argo-cd-apps/base/all-clusters/infra-deployments/tracing-workload-otel-collector/tracing-workload-otel-collector.yaml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                enabled: false
+              jaeger-thrift:
+                enabled: false
+              jaeger-grpc:
+                enabled: false
+              zipkin:
+                enabled: false
+
+      destination:
+        namespace: appstudio-otel
```
Comment (amisstea) [inline argo-cd-apps/base/all-clusters/infra-deployments/tracing-workload-otel-collector/tracing-workload-otel-collector.yaml:119]:
```suggestion
        namespace: konflux-otel
```

File: configs/enable-tekton-tracing/enable-tekton-tracing.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-tracing
+  namespace: openshift-pipelines
+data:
+  enabled: "true"
+  endpoint: "http://open-telemetry-opentelemetry-collector.appstudio-otel.svc.cluster.local:4318/v1/traces"
```
Comment (amisstea) [inline configs/enable-tekton-tracing/enable-tekton-tracing.yaml:9]:
```suggestion
  endpoint: "http://open-telemetry-opentelemetry-collector.konflux-otel.svc.cluster.local:4318/v1/traces"
```

File: components/tracing/OWNERS
Comment (mike-kingsbury) [reply components/tracing/OWNERS:1]: I'm going to leave it; I believe we'll start filling it when we configure secrets for the collector for stage & prod.

File: argo-cd-apps/base/all-clusters/infra-deployments/tracing-workload-otel-collector/tracing-workload-otel-collector.yaml (hunk 3 above)
Comment (mike-kingsbury) [reply argo-cd-apps/base/all-clusters/infra-deployments/tracing-workload-otel-collector/tracing-workload-otel-collector.yaml]: Updated...

File: configs/enable-tekton-tracing/enable-tekton-tracing.yaml (hunk 4 above)
Comment (mike-kingsbury) [reply configs/enable-tekton-tracing/enable-tekton-tracing.yaml]: Updated

**Review by amisstea — APPROVED** [review]
/lgtm

**Comment by dheerajodha** [comment]
@mike-kingsbury Hmm, I think the status reporting of these failed jobs by the openshift-ci bot is messed up. Just 7 mins ago it reported the job that failed 2 hours ago...btw [that failure](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/redhat-appstudio_infra-deployments/3740/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests/1796567450817400832#1:build-log.txt%3A1326) has already been addressed so shouldn't occur again. Let's try again.

/retest-required

**Comment by mftb** [comment]
Looks like the test failures are not (at least explicitly) related to this PR's changes. Should we keep retesting until they pass or can we safely ignore them?

**Comment by mftb** [comment]
Should we rebase this and retest @dheerajodha ?

**Comment by dheerajodha** [comment]
> Should we rebase this and retest @dheerajodha ?

No need. The required CI job (`ci/prow/appstudio-e2e-tests`) has passed now :D
It just needs somebody to add the approved and lgtm labels now.

**Comment by dheerajodha** [comment]
> It just needs somebody to add the approved and lgtm labels now.

I think @amisstea can help with that.

**Review by amisstea — APPROVED** [review]
/lgtm

**Pull request merged**